container/README.adoc

```diff
@@ -0,0 +1,302 @@
+
+The required information can be provided by using a properties file instead of independent environment variables.
+
+Create the properties file, in this document the file will be `${HOME}/.k8s-infra-container.env`.
+
+[source, shell]
+----
+$ cat << EOF > ${HOME}/.k8s-infra-container.env
+GIT_EMAIL=johndoe@redhat.com
+GIT_USERNAME=johndoe
+PASS_GPG_KEYGRIP_NAME=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+SSH_ID_KEY=my_id_github
+EOF
+----
+
+CAUTION: Adding the gpg passphrase to the file might not be such a good idea.
+
+CAUTION: Don't forget to change the file permissions to `U:R` only `chmod 600 ${HOME}/.k8s-infra-container.env`.
+
+== Generic considerations using container
+
+Using the container requires the following volumes/information to be provided:
+
+* Access to the host `.gnupg` folder:
+** e.g.: `-v ${HOME}/.gnupg:/opt/volumes/gnupg:Z`
+* Variables
+** `PASS_GPG_KEYGRIP_NAME`
+** `PASS_GPG_KEY_PASSPHRASE`
+** `GIT_EMAIL`: Email address that the commits will be associated with;
+** `GIT_USERNAME`: git user name;
+
+The following are optional volumes:
+
+* k8s-infra repository using a host folder:
+** using: `-v /path-to-k8s-infra-folder//k8s-infra:/opt/volumes/k8s-infra:Z`
+** otherwise the repository will be pulled from 
+* pass repository using a host folder:
+** e.g.: `-v /path-to-k8s-infra-folder//pass/:/opt/volumes/pass:Z`
+* Variables
+** `SSH_ID_KEY`: otherwise no ssh key will be defined;
+
+== Using the Ansible Actions
+
+The container includes the following actions:
+
+* `create_vm`: to create a new VM;
+* `remove_vm`: to remove an existing VM;
+* `k8s_install`: to deploy a k8s instance on a host;
+
+Sample environment files are included in the `samples` subfolder. They can be used when calling the `podman run` command with the `--env-file` option and have values overriden with the `-e` option.
+
+=== Create a VM
+
+Required parameters:
+
+.Required parameters for creating a VM
+[cols="1,1m,3",options="header"]
+|===
+|Variable name
+|Value(s)
+|Description
+
+|CLOUD_PROVIDER
+|[hetzner,openstack]
+|Cloud provider to be used
+
+|ANSIBLE_ACTION
+|vm_create
+|
+
+|VM_NAME
+|vm_create
+|
+
+|VM_FLAVOR
+|vm_create
+|
+
+|VM_IMAGE
+|vm_create
+|
+
+|K8S_VERSION
+|[121]
+|k8s version to be installed
+
+|===
+
+[source, shell]
+----
+$ VM_NAME=testcontainer-k121-centos8-test-04
+$ podman run --privileged -it --rm -v /path-to-k8s-infra-folder//k8s-infra:/opt/volumes/k8s-infra:Z -v /path-to-k8s-infra-folder//pass/:/opt/volumes/pass:Z -v ${HOME}/.gnupg:/opt/volumes/gnupg:Z -v $(dirname ${SSH_AUTH_SOCK}):/ssh-agent -e SSH_AUTH_SOCK="/ssh-agent" --env-file ${HOME}/.k8s-infra-container.env -v ${HOME}/.ssh:/root/.ssh:Z  -e PASS_GPG_KEY_PASSPHRASE=${PASS_GPG_KEY_PASSPHRASE} --env-file ./container/samples/vm_create.env -e CLOUD_PROVIDER=openstack -e ANSIBLE_ACTION=vm_create -e VM_NAME=${VM_NAME} -e VM_FLAVOR="ci.m5.large" -e VM_IMAGE="CentOS-8-x86_64-GenericCloud-released-latest" -e K8S_VERSION="121" localhost/k8s-infra-ansible
```

Comment:
The process to use the container image is still very complex as it will require that the user has already configured different stuffs locally and finally using this command vs `ansible playbook` of the k8s-infra ansible project could be easier.

The angle that we should follow to use this image should be different. I will detail my idea here:

Like the OpenShift coreos bot which allows to request to create an OpenShift cluster, we could instead do something like this

- Create a Java Operator + CRD able to create a K8s Job
- When a user requests to create a VM and/or to provision it with k8s or ocp, then the Operator based on the CR content (= VM CRD) will create a K8s Job
- The K8s job will use the k8s-infra-ansible-password-store image + args/env coming from the CRD to execute the command able to create or delete a VM using ansible playbook and next (if requested) to provision the VM with k8S/ocp

Advantage: The developers don't need anymore to setup GPG and GIT information as they should come from a DB (password, git, gpg keys, ...) running as a separate k8s pod or vault using the kerberos or user key passed to the CRD

So using locally a bash script or small CLI, a user could request to create/delete a VM and/or provision it with k8s/ocp, etc: 

Examples:

`hal create vm cloud=hetzner name=h01-k122 user=cmoullia`
`hal delete vm cloud=hetzner name=h01-k122`
`hal list cloud=hetzner` => to list the VMs available

WDYT: @jacobdotcosta @iocanel @BarDweller